lib/msf/core/web_services/servlet/file_servlet.rb

```diff
@@ -0,0 +1,249 @@
+      end
+      set_json_data_response(response: result)
+    }
+  end
+
+  def self.dir_mkdir
+    lambda {
+      warden.authenticate!
+      opts = parse_json_request(request, true)
+      opts_path = opts[:path] || ''
+      path = File.join(Msf::Config.rest_files_directory, opts_path)
+      if safe_expand_path?(path)
+        FileUtils.mkdir_p(path)
+        result = { path: path }
+      else
+        result = { message: 'Failed to create folder' }
+      end
+      set_json_data_response(response: result)
+    }
+  end
+
+  def self.dir_delete
+    lambda {
+      warden.authenticate!
+      opts = parse_json_request(request, true)
+      opts_path = opts[:path] || ''
+      path = File.join(Msf::Config.rest_files_directory, opts_path)
+      if !rest_files_directory?(path) && safe_expand_path?(path) && File.directory?(path)
+        FileUtils.rm_rf(path)
+        result = { path: path }
+      else
+        result = { message: 'No such directory' }
+      end
+      set_json_data_response(response: result)
+    }
+  end
+
+  def self.dir_rename
+    lambda {
+      warden.authenticate!
+      opts = parse_json_request(request, true)
+      opts_path = opts[:path] || ''
+      opts_new_path = opts[:new_path] || ''
+      path = File.join(Msf::Config.rest_files_directory, opts_path)
+      new_path = File.join(Msf::Config.rest_files_directory, opts_new_path)
+      if (!rest_files_directory?(path) && safe_expand_path?(path) && File.directory?(path)) \
+        && (!rest_files_directory?(new_path) && safe_expand_path?(new_path))
+        FileUtils.mv(path, new_path)
+        result = { path: new_path }
+      else
+        result = { message: 'No such directory' }
+      end
+      set_json_data_response(response: result)
+    }
+  end
+
+  # root
+  def self.root_path
+    lambda {
+      warden.authenticate!
+      result = { path: File.expand_path(Msf::Config.rest_files_directory) + File::SEPARATOR }
+      set_json_data_response(response: result)
+    }
+  end
+
+  # search
+  def self.search_file
+    lambda {
+      warden.authenticate!
+      sanitized_params = sanitize_params(params, env['rack.request.query_hash'])
+      opts_path = sanitized_params[:path] || ''
+      search_term = sanitized_params[:search_term] || ''
+      path = File.join(Msf::Config.rest_files_directory, opts_path)
+      search_file_paths = []
+      utf8_buf = search_term.dup.force_encoding('UTF-8')
+      # Return search keywords file path
+      if utf8_buf.valid_encoding? && safe_expand_path?(path) && File.directory?(path)
+        regex = Regexp.new(Regexp.escape(utf8_buf), true)
+        Find.find(path) do |file_path|
+          search_file_paths << file_path if (file_path =~ regex) && !File.directory?(file_path)
+        end
+        result = { path: search_file_paths }
+      else
+        result = { message: 'Invalid parameter' }
+      end
+      set_json_data_response(response: result)
+    }
+  end
+end
+
+def list_local_path(path)
```

Comment:
It looks like these functions are now global, it'd be great to move them into the servlet if possible :+1:

Reply:
I am a beginner, what does servlet mean, what should I do, thank you!

Reply:
> It looks like these functions are global, it'd be great to move them into the servlet if possible

I am a beginner, what does servlet mean, what should I do, thank you!